src/Libraries/RevitNodes/GeometryConversion/CurveLoopPartition.cs

```diff
@@ -0,0 +1,126 @@
+            foreach (var curveLoop in curveLoops)
+            {
+                var verts = TesselateCurveLoop(curveLoop).ToList();
+                List<Autodesk.DesignScript.Geometry.Curve> curves =
+                    new List<Autodesk.DesignScript.Geometry.Curve>();
+                foreach (var curve in curveLoop)
+                {
+                    curves.Add(curve.ToProtoType(false));
+                }
+                var geom = new List<Autodesk.DesignScript.Geometry.Geometry>();
+                geom.Add(PolyCurve.ByJoinedCurves(curves));
+                tesselatedCurveLoops.Add(new Tuple<BoundingBox, List<XYZ>, List<Curve>>(BoundingBox.ByGeometry(geom), verts, curveLoop));
+                curves.ForEach(x => x.Dispose());
+                curves.Clear();
+                geom.ForEach(x => x.Dispose());
+                geom.Clear();
+            }
+
+            var outerCurveLoops = new List<Tuple<BoundingBox, List<XYZ>, List<Curve>>>();
+            var innerCurveLoops = new List<Tuple<BoundingBox, List<XYZ>, List<Curve>>>();
+
+            // collect the outer loops, which are not contained in any other loop
+            for (int i = 0, j = tesselatedCurveLoops.Count - 1; i < tesselatedCurveLoops.Count; j = i++) 
+            {
+                var curveLoop = tesselatedCurveLoops[i];
+                var isOuter = tesselatedCurveLoops.Where((_, l) => i != l)
+                    .All(bound => !curveLoop.Item2.IsContainedIn(bound.Item1));
+
+                if(isOuter)
+                {
+                    outerCurveLoops.Add(curveLoop);
+                }
+                else
+                {
+                    innerCurveLoops.Add(curveLoop);
+                }
+            }
+
+            var partitionedLoops = new List<List<List<Curve>>>();
+
+            // for each outer loop, collect the loops that are inside of it
+            // forming the partitions of the original curve loop set
+            foreach (var outerLoop in outerCurveLoops)
+            {
+                var comp = new List<List<Curve>> { outerLoop.Item3 };
+                var mask = new List<bool>();
+                foreach (var innerLoop in innerCurveLoops)
+                {
+                    if (innerLoop.Item2.IsContainedIn(outerLoop.Item1))
+                    {
+                        mask.Add(false);
+                        comp.Add(innerLoop.Item3);
+                    }
+                    else
+                    {
+                        mask.Add(true);
+                    }
+                }
+                // remove innerEdge loops that have already been added to a partition
+                innerCurveLoops = innerCurveLoops.Where((_, j) => mask[j]).ToList();
+                // add the new partition
+                partitionedLoops.Add(comp);
+            }
+
+            return partitionedLoops;
+        }
+
+        public static List<List<Autodesk.Revit.DB.Curve>> GetAllCurveloopsFromRevitFace(Face face)
```

Comment:
Return nested lists makes the function unclear, I suggest that just return CurveLoop.